python/modules/map_plate.py

```diff
@@ -78,49 +113,129 @@ def _get_param_paths(input_path):
+    """
+    Function gettin all values for a given param 
+    from map_plate dictionary.
+    """
     values = []
     for well in mp_dict:
         values.append(mp_dict[well][param])
     return values
 
-def get_param_unique_values(mp_dict, param): # get all unique values for a given param from map_plate dictionary
-    values = get_param_all_values(mp_dict, param)
+def get_param_unique_values(mp_dict, param): 
+    """
+    Function gettin all unique values for a given param 
+    from map_plate dictionary.
+    """
+    values = get_param_values(mp_dict, param)
     unique = list(set(values))
     return unique
 
-def get_all_params_names(mp_dict): # get all key names from map_plate dictionary
+def get_params_names(mp_dict): # get all key names from map_plate dictionary
     names = []
     key_0 = list(mp_dict.keys())[0]
-    names = list(mp_dict[key_0].keys()) # if we assume that number of information between wells is constant
+    names = list(mp_dict[key_0].keys()) 
+    """
+    Uncommented option (above):
+        if we assume that number of information between wells is constant
+    Commented option (below): 
+        if we assume that number of information about wells may somehow differ
+    """
     """params_count = len(mp_dict[key_0])
-    for well in mp_dict: # if we assume that number of information about wells may somehow differ
+    for well in mp_dict: 
         tmp= len(mp_dict[well])
         if params_count != tmp:
-            logger.warning("Some wells are not described by the same number of information: %s, %s", key_0, well)
+            logger.warning("Some wells are not described by " 
+                    "the same number of information: %s, %s", key_0, well)
             if tmp > params_count:
                 names = []
                 params_count = tmp
                 for param in mp_dict[well]:
                     names.append(param)"""
     return names
 
-def get_well_params(mp_dict, well): # get all parameters values for a given well
+def get_well_params(mp_dict, well):
+    """ 
+    Get all parameters values for a given well
+    """
     values = []
     for param in mp_dict[well]:
         values.append(mp_dict[well][param])
     return values
+
+def _verify_input_mp_file(input_file, delimiter, dimensions = 0):
+    if not FM.file_verify_extension(input_file, ".csv"):
+        logger.error("Wrong input format: %s", input_file)
+        return False
+    if dimensions != 0:
+        data = CSV_M.read_csv(input_file, delimiter)
+        x = len(data)
+        y = len(data[0])
+        if dimensions != [x,y]:
+            logger.error("Wrong input file "
+                        "(improper csv dimensions): %s.", input_file)
+            return False
+    return True
+
+
+def _prepare_output(input_data, mp_dict, compare_column):
+    output_data = []
+    column_names = input_data[0]
+    for i in range(len(column_names)):
+        if column_names[i] == compare_column:
+            position = i
```

Comment:
@stork119 
Software is looking for column with given name to extract wells name from it to compare. This function was created for map_plate working after csv_merge (before we discussed problem). I'm gonna change it when new task_parallelize (based on map_plate) will be ready.